### ISSUE TITLE
Reserved word and uniqueness validation

### ISSUE THREAD

**Issue by chubchenko** [djezzzl/database_consistency#72]
First of all, I want to say thank you for this gem!!!

We found an issue where the column name is equal to a reserved word and we have to add an index to make it unique.

```ruby
RSpec.describe DatabaseConsistency::Checkers::MissingUniqueIndexChecker do
  subject(:checker) { described_class.new(model, attribute, validator) }

  let(:model) { klass }
  let(:attribute) { :from }
  let(:validator) { klass.validators.first }

  context 'with postgresql database' do
    include_context 'postgresql database context'

    context 'when uniqueness validation has case sensitive option turned off' do
      context 'when the column name is equal to a reserved word' do
        let(:klass) { define_class { |klass| klass.validates :from, uniqueness: { case_sensitive: false } } }

        before do
          define_database_with_entity do |table|
            table.string :from
            table.index "lower('from')", unique: true
          end
        end

        specify do
          expect(checker.report).to have_attributes(
            checker_name: 'MissingUniqueIndexChecker',
            table_or_model_name: klass.name,
            column_or_attribute_name: 'lower(from)',
            status: :ok,
            message: nil
          )
        end
      end
    end
  end
end
```

It looks like the problem occurs during the following comparison:

```ruby
extract_index_columns(index.columns).sort == sorted_index_columns
```
which equals
```ruby
["lower('from'::text)"] == ["lower(from)"]
```

**Comment by djezzzl** [comment]
Hey @chubchenko , thank you for pointing this out! And for using the gem, of course!

I'm not sure how smart should we be here, but I guess it would be fine if compare assuming that value can be casted at any type. So, I'll try to fix it ASAP 👍

**Comment by djezzzl** [comment]
The fix is here: https://github.com/djezzzl/database_consistency/pull/73. I'll release it probably today (with some other improvements).

**Comment by djezzzl** [comment]
Hey, I just released `0.8.9`, please try it out and let me know if it didn't help. Feel free to reopen the issue if needed.

P.S. Sorry it took me so long to release.
